### PR TITLE
API contract for discord API

### PR DESCRIPTION
**Pull request by vinayak-trivedi**
This PR adds API contract for discord API for generating and getting discord link API.

There are two APIs:

`GET /discord-actions/invite`: Gets previously generated link for a user
`POST /discord-actions/invite`: Generates link for a particular user.

### CODE REVIEW

File: discord-actions/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+---
+
+## **GET /discord-actions/invite**
+
+- return the discord invite created for the user
+
+- **Query**
+
+  - Optional: `userId=[string]`: if this is provided it returns the invite of the user whose userId is provided in the query param, if the user is not a super user then this should be equal to the userId of the user, if no userId is provided, it return invite for the user who has made the request
```
Comment (pallabez) [inline discord-actions/README.md:18]: Whats the use of this? I'm not getting it

Comment (vinayak-trivedi) [reply discord-actions/README.md:18]: This is done in case the super user wants to see the link generated for a particular user

**Review by pallabez — APPROVED** [review]
LGTM

File: discord-actions/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      "statusCode": 500,
+      "error": "Internal Server Error",
+      "message": "Something went wrong. Please contact admin"
+    }
+    ```
+
+## **POST /discord/invites**
+
+- This Api is used to create a discord invite for the user, invite for a particular user can be generated by a super user or a that user only, invite for a particular user can be created only once, Following conditions will be there for creating invite for the user:
+- The user should not be archieved
```
Comment (satyam73) [inline discord-actions/README.md:77]:
```suggestion
- The user should not be archived
```

**Review by satyam73 — APPROVED** [review]

File: discord-actions/README.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      }
+      ```
+
+  - **Code:** 403
+    - **Content:**
+      ```json
+        {
+          "statusCode": 403,
+          "error": "Forbidden",
+          "message": "User should be super user to generate link for other users" "||" "Only users who have never joined discord can generate invite link" "||" "Archived users cannot generate invite" "||" "Only selected roles can generate discord link directly"
```
[the remark this answers is not in the document]
Comment (vinayak-trivedi) [reply discord-actions/README.md:125]: This is changed permanently or temporarily, if this is temporarily done, we can get this merged with a note with it, that temporarily this is disabled.

[the remark this answers is not in the document]
Comment (vinit717) [reply discord-actions/README.md:125]: Yes this is temporary

**Review by vinit717 — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]

File: discord-actions/README.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      ```json
+      {
+        "statusCode": 500,
+        "error": "Internal Server Error",
+        "message": "An internal server error occurred"
+      }
+      ```
+    ```
+
+    ```
```
Comment (prakashchoudhary07) [inline discord-actions/README.md:141]: Can you please remvoe this extra block?

**Pull request merged**